### PR TITLE
AC: Added data analyzer for segmentation tasks

### PR DESCRIPTION
**Pull request by Anna-Mironova**

### CODE REVIEW

File: tools/accuracy_checker/accuracy_checker/data_analyzer/coco_instance_segmentation_analyzer.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        data_analysis['total_instances'] = total_instances
+        label_map = meta.get('label_map', {})
+
+        for key in counter:
+            if key in label_map:
+                print_info('{name}: {value}'.format(name=label_map[key], value=counter[key]))
+                data_analysis[label_map[key]] = counter[key]
+            else:
+                print_info('class_{key}: {value}'.format(key=key, value=counter[key]))
+                data_analysis['class_{key}'.format(key=key)] = counter[key]
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/data_analyzer/coco_instance_segmentation_analyzer.py:46]:
```suggestion
           class_name = label_map.get(key, 'class_{}'.format(key))
           print_info('{class_name}: {value}'.format(class_name=class_name, value=counter[key]))
           data_analysis[class_name] = counter[key]
```

File: tools/accuracy_checker/accuracy_checker/data_analyzer/coco_instance_segmentation_analyzer.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                if areas.get(label):
+                    areas[label]['area'].append(float(rect[2] * rect[3]))
+                    areas[label]['width'].append(float(rect[2]))
+                    areas[label]['height'].append(float(rect[3]))
+                else:
+                    areas[label]={'area': [float(rect[2] * rect[3])], 'width': [float(rect[2])], 'height': [float(rect[3])]}
+
+        for key in areas:
+            size = counter[key]
+            areas[key]['area'] = sum(areas[key]['area']) / size
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/data_analyzer/coco_instance_segmentation_analyzer.py:46]: probably average for all 3 parameters, min/max also useful info

File: tools/accuracy_checker/accuracy_checker/data_analyzer/coco_instance_segmentation_analyzer.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            areas[key]['width'] = sum(areas[key]['width']) / size
+            areas[key]['height'] = sum(areas[key]['height']) / size
+
+        print_info('Total instances: {value}'.format(value=total_instances))
+        data_analysis['total_instances'] = total_instances
+        label_map = meta.get('label_map', {})
+
+        for key in counter:
+            if key in label_map:
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/data_analyzer/coco_instance_segmentation_analyzer.py:55]: the same note about code duplication when difference only in one variable

File: tools/accuracy_checker/accuracy_checker/data_analyzer/coco_instance_segmentation_analyzer.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+        counter = Counter()
+        total_instances = 0
+        characteristics = {}
+        for data in result:
+            total_instances += data.size
+            counter.update(data.labels)
+            for label, rect in zip(data.labels, data.metadata['rects']):
+                if characteristics.get(label):
+                    characteristics[label]['area'].append(float(rect[2] * rect[3]))
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/data_analyzer/coco_instance_segmentation_analyzer.py:38]: area is not always equal area of box, please use `data.area` for getting correct area of mask

File: tools/accuracy_checker/accuracy_checker/data_analyzer/segmentation_analyzer.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        return annotation
+
+    def analyze(self, result: list, meta, count_objects=True):
+        data_analysis = {}
+        if count_objects:
+            data_analysis['annotations_size'] = self.object_count(result)
+
+        counter = Counter()
+
+        annotations = deepcopy(result)
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/data_analyzer/segmentation_analyzer.py:48]: better run copy for instance, not for all list at the time. Representation storage can require a lot of memory, so copy whole list can increase  RAM usage dramatically

File: tools/accuracy_checker/accuracy_checker/data_analyzer/segmentation_analyzer.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+        for data in annotations:
+            data.set_segmentation_mask_source(meta['segmentation_masks_source'])
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/data_analyzer/segmentation_analyzer.py:51]: also can be done in the same cycle with pixels counting

File: tools/accuracy_checker/accuracy_checker/data_analyzer/segmentation_analyzer.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+        segmentation_colors = meta.get('segmentation_colors')
+        if segmentation_colors:
+            annotations = self._encode_mask(annotations, segmentation_colors)
+
+        for data in annotations:
+            for elem in data.mask:
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/data_analyzer/segmentation_analyzer.py:58]: just imagine, you have dataset with 200 images 1024x1280 resolution
on each image you will perfom this 1024 times on each image, it consumes a lot of time...

better to use np.unique for counting unique pixel values and their frequencies

**Comment by Anna-Mironova** [comment]
Jenkins please retry a build

**Comment by Anna-Mironova** [comment]
Jenkins please retry a build

File: tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py [hunk 8]
```diff
@@ -102,7 +102,7 @@ def configure(self):
     def convert(self, check_content=False, progress_callback=None, progress_interval=100, **kwargs):
         full_annotation = read_json(self.annotation_file)
         image_info = full_annotation['images']
-        image_ids = [(image['id'], image['file_name']) for image in image_info]
+        image_ids = [(image['id'], image['file_name'], np.array([image['height'], image['width'], None])) for image in image_info]
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py:105]:
```suggestion
        image_ids = [(image['id'], image['file_name'], np.array([image['height'], image['width'], 3])) for image in image_info]
```

**Pull request merged**